Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -397,7 +397,7 @@ void WritePixelShaderCommonHeader(ShaderCode& out, APIType ApiType, u32 num_texg
             "\tint4 " I_INDTEXMTX "[6];\n"
             "\tint4 " I_FOGCOLOR ";\n"
             "\tint4 " I_FOGI ";\n"
-            "\tfloat4 " I_FOGF "[2];\n"
+            "\tfloat4 " I_FOGF "[5];\n"
```

Comment:
might you please use a "float fog[12];" instead? This will help with the addressing, and even std140 will pack this.